### ISSUE TITLE
quickstart does not work for mac machines

### ISSUE THREAD

**Issue by bobzhang** [nyergler/hieroglyph#94]
tmp$ hieroglyph-quickstart 
hieroglyph-quickstart 
Welcome to the Hieroglyph 0.7 quickstart utility.

This will ask questions for creating a Hieroglyph project, and then ask
some basic Sphinx questions.

The presentation title will be included on the title slide.
> Presentation title: x
x
> Author name(s): t
t
Traceback (most recent call last):
  File "/usr/local/bin/hieroglyph-quickstart", line 9, in <module>
    load_entry_point('hieroglyph==0.7', 'console_scripts', 'hieroglyph-quickstart')()
  File "/usr/local/lib/python2.7/site-packages/hieroglyph/quickstart.py", line 155, in quickstart
    ask_user(d)
  File "/usr/local/lib/python2.7/site-packages/hieroglyph/quickstart.py", line 119, in ask_user
    for t in theme_entrypoints
AttributeError: 'EntryPoint' object has no attribute 'resolve'

**Comment by nyergler** [comment]
Thanks for reporting the issue, I think this is fixed on `master`; would you mind installing the development version to try it? You can run:

```
$ pip install -U hieroglyph==dev --allow-external hieroglyph --allow-unverified hieroglyph
```

And then try `hieroglyph-quickstart` again.

I'll get a dot release out shortly, but it'd be great to have this verified.

Thanks!

**Comment by bobzhang** [comment]
failed to install

Installing collected packages: hieroglyph, setuptools, Sphinx, sphinx-rtd-theme, alabaster, babel, Pygments, snowballstemmer, pytz
  Found existing installation: hieroglyph 0.7
    Uninstalling hieroglyph:
      Successfully uninstalled hieroglyph
  Running setup.py install for hieroglyph
    
    Installing hieroglyph-quickstart script to /usr/local/bin
    Installing hieroglyph script to /usr/local/bin
  Found existing installation: setuptools 5.4.2
    Uninstalling setuptools:
      Successfully uninstalled setuptools
  Rolling back uninstall of setuptools
Cleaning up...
Exception:
Traceback (most recent call last):
  File "/usr/local/Cellar/python/2.7.8_2/Frameworks/Python.framework/Versions/2.7/lib/python2.7/site-packages/pip-1.5.6-py2.7.egg/pip/basecommand.py", line 122, in main
    status = self.run(options, args)
  File "/usr/local/Cellar/python/2.7.8_2/Frameworks/Python.framework/Versions/2.7/lib/python2.7/site-packages/pip-1.5.6-py2.7.egg/pip/commands/install.py", line 283, in run
    requirement_set.install(install_options, global_options, root=options.root_path)
  File "/usr/local/Cellar/python/2.7.8_2/Frameworks/Python.framework/Versions/2.7/lib/python2.7/site-packages/pip-1.5.6-py2.7.egg/pip/req.py", line 1439, in install
    requirement.rollback_uninstall()
  File "/usr/local/Cellar/python/2.7.8_2/Frameworks/Python.framework/Versions/2.7/lib/python2.7/site-packages/pip-1.5.6-py2.7.egg/pip/req.py", line 603, in rollback_uninstall
    self.uninstalled.rollback()
  File "/usr/local/Cellar/python/2.7.8_2/Frameworks/Python.framework/Versions/2.7/lib/python2.7/site-packages/pip-1.5.6-py2.7.egg/pip/req.py", line 1855, in rollback
    pth.rollback()
AttributeError: 'str' object has no attribute 'rollback'

**Comment by bobzhang** [comment]
Full log


slides>pip install -U hieroglyph==dev --allow-external hieroglyph --allow-unverified hieroglyph
Downloading/unpacking hieroglyph==dev
  hieroglyph an externally hosted file and may be unreliable
  hieroglyph is potentially insecure and unverifiable.
  Downloading master (unknown size): 2.1MB downloaded
  Running setup.py (path:/private/var/folders/df/dh8jkb254qv48cfc0_rhmkf00000gn/T/pip_build_hongbozhang/hieroglyph/setup.py) egg_info for package hieroglyph
    
  Requested hieroglyph==dev, but installing version 0.7.1.dev
Downloading/unpacking setuptools>=11.3 (from hieroglyph==dev)
  Downloading setuptools-14.3.1-py2.py3-none-any.whl (501kB): 501kB downloaded
Downloading/unpacking Sphinx>=1.2 from https://pypi.python.org/packages/2.7/S/Sphinx/Sphinx-1.3.1-py2.py3-none-any.whl#md5=9abda702fcb41e475b9771fbb4bfe371 (from hieroglyph==dev)
  Downloading Sphinx-1.3.1-py2.py3-none-any.whl (1.3MB): 1.3MB downloaded
Requirement already up-to-date: six in /usr/local/lib/python2.7/site-packages (from hieroglyph==dev)
Downloading/unpacking sphinx-rtd-theme>=0.1,<0.2 (from Sphinx>=1.2->hieroglyph==dev)
  Downloading sphinx_rtd_theme-0.1.7-py2-none-any.whl (220kB): 220kB downloaded
Requirement already up-to-date: Jinja2>=2.3 in /usr/local/lib/python2.7/site-packages (from Sphinx>=1.2->hieroglyph==dev)
Downloading/unpacking alabaster>=0.7,<0.8 (from Sphinx>=1.2->hieroglyph==dev)
  Downloading alabaster-0.7.3.tar.gz
  Running setup.py (path:/private/var/folders/df/dh8jkb254qv48cfc0_rhmkf00000gn/T/pip_build_hongbozhang/alabaster/setup.py) egg_info for package alabaster
    
    warning: no files found matching 'README.md'
    warning: no previously-included files matching '.*.swp' found under directory 'alabaster'
Downloading/unpacking babel>=1.3 (from Sphinx>=1.2->hieroglyph==dev)
  Downloading Babel-1.3.tar.gz (3.4MB): 3.4MB downloaded
  Running setup.py (path:/private/var/folders/df/dh8jkb254qv48cfc0_rhmkf00000gn/T/pip_build_hongbozhang/babel/setup.py) egg_info for package babel
    
    warning: no previously-included files matching '*' found under directory 'docs/_build'
    warning: no previously-included files matching '*.pyc' found under directory 'tests'
    warning: no previously-included files matching '*.pyo' found under directory 'tests'
Downloading/unpacking Pygments>=2.0 (from Sphinx>=1.2->hieroglyph==dev)
  Downloading Pygments-2.0.2-py2-none-any.whl (672kB): 672kB downloaded
Downloading/unpacking snowballstemmer>=1.1 (from Sphinx>=1.2->hieroglyph==dev)
  Downloading snowballstemmer-1.2.0.tar.gz (49kB): 49kB downloaded
  Running setup.py (path:/private/var/folders/df/dh8jkb254qv48cfc0_rhmkf00000gn/T/pip_build_hongbozhang/snowballstemmer/setup.py) egg_info for package snowballstemmer
    
    warning: no files found matching '*.py' under directory 'src'
Requirement already up-to-date: docutils>=0.11 in /usr/local/lib/python2.7/site-packages (from Sphinx>=1.2->hieroglyph==dev)
Requirement already up-to-date: markupsafe in /usr/local/lib/python2.7/site-packages (from Jinja2>=2.3->Sphinx>=1.2->hieroglyph==dev)
Downloading/unpacking pytz>=0a (from babel>=1.3->Sphinx>=1.2->hieroglyph==dev)
  Downloading pytz-2015.2-py2.py3-none-any.whl (476kB): 476kB downloaded
Installing collected packages: hieroglyph, setuptools, Sphinx, sphinx-rtd-theme, alabaster, babel, Pygments, snowballstemmer, pytz
  Found existing installation: hieroglyph 0.7
    Uninstalling hieroglyph:
      Successfully uninstalled hieroglyph
  Running setup.py install for hieroglyph
    
    Installing hieroglyph-quickstart script to /usr/local/bin
    Installing hieroglyph script to /usr/local/bin
  Found existing installation: setuptools 5.4.2
    Uninstalling setuptools:
      Successfully uninstalled setuptools
  Rolling back uninstall of setuptools
Cleaning up...
Exception:
Traceback (most recent call last):
  File "/usr/local/Cellar/python/2.7.8_2/Frameworks/Python.framework/Versions/2.7/lib/python2.7/site-packages/pip-1.5.6-py2.7.egg/pip/basecommand.py", line 122, in main
    status = self.run(options, args)
  File "/usr/local/Cellar/python/2.7.8_2/Frameworks/Python.framework/Versions/2.7/lib/python2.7/site-packages/pip-1.5.6-py2.7.egg/pip/commands/install.py", line 283, in run
    requirement_set.install(install_options, global_options, root=options.root_path)
  File "/usr/local/Cellar/python/2.7.8_2/Frameworks/Python.framework/Versions/2.7/lib/python2.7/site-packages/pip-1.5.6-py2.7.egg/pip/req.py", line 1439, in install
    requirement.rollback_uninstall()
  File "/usr/local/Cellar/python/2.7.8_2/Frameworks/Python.framework/Versions/2.7/lib/python2.7/site-packages/pip-1.5.6-py2.7.egg/pip/req.py", line 603, in rollback_uninstall
    self.uninstalled.rollback()
  File "/usr/local/Cellar/python/2.7.8_2/Frameworks/Python.framework/Versions/2.7/lib/python2.7/site-packages/pip-1.5.6-py2.7.egg/pip/req.py", line 1855, in rollback
    pth.rollback()
AttributeError: 'str' object has no attribute 'rollback'

Storing debug log for failure in /Users/hongbozhang/.pip/pip.log

**Comment by bobzhang** [comment]
If I have sphinx installed, is it possible to get all dependencies distributed in a package?

**Comment by jldiaz** [comment]
It's not only a Mac issue. I had the same problem in Ubuntu 14.04 LTS. The dev version solves it.

**Comment by nyergler** [comment]
@bobzhang I notice you're using Python from Homebrew; does it work any differently using the system Python? I don't use Homebrew's Python on OS X, so hadn't considered testing with that. It's not at all clear to me why it's rolling back (or why the rollback fails); I sort of suspect it has to do with the `setuptools` upgrade.

@jldiaz Thanks for the update on Ubuntu.

I'll try to get a 0.7.1 release out this weekend.

**Comment by nyergler** [comment]
Released 0.7.1, which contains a fix for this. 

Thanks again for reporting.
